node/src/utils/ds.rs

```diff
@@ -63,13 +63,15 @@ where
     if vec.len() < SAMPLE_SIZE {
         vec.estimate_heap_size()
     } else {
+        let base_size = vec.capacity() * mem::size_of::<T>();
+
         let mut rng = sampling_rng(vec.len());
         let sampled = vec
             .as_slice()
             .choose_multiple(&mut rng, SAMPLE_SIZE)
-            .map(DataSize::estimate_heap_size)
+            .map(|v| v.estimate_heap_size())
```

Comment:
I thought that was exactly equivalent to `.map(DataSize::estimate_heap_size)`?? Don't you have to `(*v).estimate_heap_size()` or something?